packages/rum-core/src/domain/assembly.ts

```diff
@@ -79,10 +84,22 @@ export function startRumAssembly(
             type: getSessionType(),
           },
         }
-        const serverRumEvent = (needToAssembleWithAction(rawRumEvent)
+        let serverRumEvent = (needToAssembleWithAction(rawRumEvent)
           ? combine(rumContext, viewContext, actionContext, rawRumEvent)
           : combine(rumContext, viewContext, rawRumEvent)) as RumEvent & Context
 
+        if (needToAssembleWithInForeground(rawRumEvent)) {
+          serverRumEvent = combine(serverRumEvent, inForegroundContext)
+        }
+
+        if (isView(rawRumEvent)) {
+          const in_foreground_periods = foregroundContexts.getInForegroundPeriods(
+            startTime,
+            rawRumEvent.view.time_spent
+          )
+          serverRumEvent = combine(serverRumEvent, { view: { in_foreground_periods } })
+        }
```

Comment:
I tried to do
```javascript
serverRumEvent.view.in_foreground_periods = in_foreground_periods
```
I though it was cleaner but the property is written as `readonly`

Reply:
Not sure what is the take away here.

Reply:
you should add it to `rawRumEvent` instead of `serverRumEvent`

Reply:
Using `foregroundContexts` in `viewCollection`, `actionCollection` and `errorCollection` instead of `assembly` could make that more natural.
It would also avoid the conversion from serverDuration to duration which feel fishy.
wdyt?